### DISCUSSION TITLE
Allow for clearer install documentation and screenshot options

### DISCUSSION THREAD

**Discussion by spaced-out-mvc** [community/community#43806] [Repositories]
### Select Topic Area

Product Feedback

### Body

Do you think it would be possible to improve github to make it clear whether repo owners have added installer documentation and a screenshot for their softwares in final working condition.

As it wastes so much users/developers time when people do not add the basics.

I think it would be good if github could highlight this at the top of every repository page to highlight whether the user has added installer documentation or a screenshot.

As on github there are millions of repos which are just blank or have incomplete readme files.
You should make a clear option to add this information and make it very clear when users have not added this information.

I think this would save millions of developers around the world plenty of time, instead of wasting their time over and over with faulty undocumented, non screenshot applications and libraries all of the damn time!!!

This information would also be good if added to the search results, so users/developers can see from the search results whether it is worth clicking on the link or not.

Without any screenshot of these applications makes it so every developer has to setup the application with no  documentation and try to produce an on screen example each time.

This has wasted so much of my time over the years and i am sure it wastes every developers time especially when half of the applications on github do not work anyway. 

Countless hours wasted. please improve this for everybody!

You should create a simple github readme checklist, and validator to check the following:

1 - Install Instructions (Yes/No)
2 - Screenshot of application (Yes/No)
3 - Unit tested (Yes/No)

Then on the screen you could give users a certificate or logo to show that they have done the basics.

Also it would be very useful if you could make it so users can use a basic readme template showing this information.
then most of github can be more organised and the readmes can start to look more similar and formal for once.
Helping people with the basics can go a very long way in helping them to be more organised and formal about their readme documentations

This would save the whole world thousands of wasted hours, please can you update github website to do this please?

**Comment by DSDmark** [comment]
Hi @space-mvc,

There are already certain community guidelines for this repository, but not many maintainers follow them. I agree that it is both time-saving and instructive for other users. All users must, at the very least, follow these community guidelines in order to use GitHub. Should refrain from building such a pointless garbage repository. 

![2023-01-10_12-27](https://user-images.githubusercontent.com/85744816/211474271-b6ecc4c1-7cdc-4f53-af5a-4d7a3efaa215.png)

**Reply by spaced-out-mvc** [reply]
I think as the readme.md is a free text document, i think it would be better to add new options seperate to the readme.md for this information.
Something you can click on, via github which says 'add screenshot' would be great. and 'add install instructions' and 'add unit test file'

Then this can be more throughly checked automatically on github.

I would say the free text readme.md alone is not enough. And if github could give users a certificate/logo/trophy for adding this information people would more likely use it, especially as users would prefer to choose a basic certified application over an non basic certified application i would think.

**Reply by spaced-out-mvc** [reply]
You are right nobody reads the guidelines, so it would be better to add these options as buttons to click on seperate to the readme.md file, then users will not need to read the guidelines as it would be a option right in front of them all the time everytime they look at their repo.

**Comment by spaced-out-mvc** [comment]
Something simple like this on the top right of this image would be perfect

![example1](https://user-images.githubusercontent.com/910208/211479400-e715ecd2-f78f-4982-9061-24bea9a096da.png)

**Reply by spaced-out-mvc** [reply]
I think this would save millions of development and test setup hours for everybody...
